unyt/tests/test_array_functions.py

```diff
@@ -1654,11 +1654,16 @@ def test_apply_along_axis():
     assert ret.units == cm**2
 
 
-def test_apply_over_axes():
+@pytest.mark.parametrize("axes, expected_units", [((0, 1), cm**4), ((0,), cm**2)])
+def test_apply_over_axes(axes, expected_units):
+    # the user-supplied function must be trusted to treat units
+    # sensibly (mainly that it doesn't give a mix of units across
+    # the resulting array), but we can check that units are
+    # propagated correctly for well-behaved functions.
     a = np.eye(3) * cm
-    ret = np.apply_over_axes(lambda x, axis: x * cm, a, (0, 1))
-    assert type(ret) is unyt_array
-    assert ret.units == cm**3
+    ret = np.apply_over_axes(lambda x, axis: x[axis] ** 2, a, axes)
+    assert isinstance(ret, unyt_array)  # could be subclass unyt_quantity
```

Comment:
Can you elaborate on this comment ? I don't think I understand what situation you're hinting at

Reply:
`assert type(ret) is unyt_array` breaks if `ret` is a `unyt_quantity`, which it is (correctly) in one of the test cases.

Reply:
oh, I forgot that we had two classes there 🤦🏻‍♂️ 
in that case, would it make sense to rewrite the comment as follow ?
```suggestion
    assert isinstance(ret, unyt_array)  # could be unyt_quantity
```

Reply:
That subclass has caused me considerable consternation (until deciding to handle it properly), it is not easily forgotten ;)